### ISSUE TITLE
LFortran does not link properly on Ubuntu

### ISSUE THREAD

**Issue by konradha** [lfortran/lfortran#715]
For any test or example that compiles with `gfortran`, LFortran yields 

```bash
sh: 1: clang: not found
The command 'clang -o a.out a.out.tmp.o  -L"/home/konrad/code/forks/lfortran/src/bin/../runtime" -Wl,-rpath,"/home/konrad/code/forks/lfortran/src/bin/../runtime" -llfortran_runtime -lm' failed.
```

I am on `Ubuntu 22.04.1 LTS` with kernel `5.15.0-47-generic` 
I think the issue lies with Ubuntu's handling of clang -- `which clang` is empty; whereas `which clang-11` yields a path depending on environment. LFortran's driver calls `clang` though.

**Comment by konradha** [comment]
After `mamba install clang` it apparently DID install the corresponding clang but the error becomes more involved

```bash
/opt/anaconda3/envs/lf/bin/../lib/gcc/x86_64-conda-linux-gnu/12.1.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/konrad/code/forks/lfortran/src/bin/../runtime/liblfortran_runtime.so: undefined reference to `clock_gettime@GLIBC_2.17'
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
The command 'clang -o arr arr.tmp.o  -L"/home/konrad/code/forks/lfortran/src/bin/../runtime" -Wl,-rpath,"/home/konrad/code/forks/lfortran/src/bin/../runtime" -llfortran_runtime -lm' failed.
```

**Comment by Thirumalai-Shaktivel** [comment]
Why not use `g++` for linking? (`--link-with-gcc`) Does this yield the same issue?

**Comment by konradha** [comment]
@Thirumalai-Shaktivel well, I think the default option should just work out of the box.

**Comment by certik** [comment]
Yes, the default should work out of the box. I believe this was changed in some of the PRs for cross-compiling.

I think we need a way to select which compiler should do the linking. Or perhaps it's time to get the LLD linker from LLVM working and just use that.

**Comment by j-bm** [comment]
New to lfortran here.
Building on Mint/Ubuntu required telling cmake to use CMAKE_CXX_COMPILER=clang++-14 not clang++, and WITH_LLVM=ON.
Perhaps the built lfortran executable could adopt the cmake definition and not rely on "clang" or "clang++" to be universal.

Another fix would be add symbol LFORTRAN_CC where lfortran uses the value as the desired C compiler.  Even tcc might be usable.  The default would be clang.

A short-term fix is to put ~/bin in the path and link /usr/bin/clang-14 to ~/bin/clang.

**Comment by certik** [comment]
Thanks @j-bm. Can you open a dedicated issue for this? If you know how to fix it in our cmake, please also go ahead and submit a PR. The `clang` for linking is hardwired in `lfortran.cpp`. You'll find there code like this:
```c++
            if (link_with_gcc) {          
                CC = "gcc";               
            } else {                                        
                CC = "clang";            
            }                                                   
                                                                         
            char *env_CC = std::getenv("LFORTRAN_CC");
            if (env_CC) CC = env_CC;
```
So it already should work with `LFORTRAN_CC`.
